### DISCUSSION TITLE
Why doesn't Inertia use something like Vuex or Pinia under the hood? (instead of data-page)

### DISCUSSION THREAD

**Discussion by AliGh1** [inertiajs/inertia#1285] [Ideas]
Hi, I'm just getting to know and working with Inertia, so I have a question, **why doesn't** use something like **Vuex** or **Pinia** instead of _**data-page**_ attribute **under the hood**?

### I'm just wondering if this is possible?

Wouldn't it look better if Inertia did it automatically under the hood?

So, we couldn't have a big _**data-page**_  attribute.

**Comment by craigrileyuk** [comment]
How do you think Vuex and Pinia get their data?

How do you propose maintaining different persistent-store delivery systems for every frontend framework? Should there be the @inertiajs/inertia-vue3-pinia branch, the @inertiajs/inertia-vue3-vuex branch, the @inertiajs/inertia-vue2-vuex branch, the @inertiajs/inertia-react-redux branch and so on...

Frontend clients are Javascript-based. Laravel outputs HTML.

For the page data to get to the Javascript frontend, it **has** to be written somewhere in the HTML.

The only other alternative is to request it via AJAX which is a huge performance penalty to make the page source look a little nicer.
